soc/nordic/nrf54h/ironside_boot_report/Kconfig

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2025 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_NRF54H20_IRONSIDE_BOOT_REPORT
+	bool "Nordic IRONside SE boot report"
+	default y if SOC_NRF54H20_CPUAPP
+	depends on SOC_NRF54H20_IRON
+	help
+	  This option enables parsing of the Boot Report populated by Nordic IRONside SE.
+
+config SOC_NRF54H_IRONSIDE_BOOT_REPORT_MAGIC
+	hex
+	default 0xe176928d
+	help
+	  Constant used to check if an Nordic IRONside SE boot report has been written.
```

Comment:
Why have a constant in the configuration system?

Wouldn't it be easier for a programmer to know that it is constant if it were a define in a header file?

Reply:
We have historically been doing it this way. I guess the idea is that these values are configurations in themselves, hence they should be in kconfig.

Also, the details of their value is of no interest of the developer. Hence it is better to not clutter the .h and .c files with this "useless" information.

Lastly it allows a description of the value itself in case that is important. For instance if it has some special properties (for instance hamming distance to other values, max number of 0's in its binary representation etc) that would otherwise clutter the .c and .h files.